### ISSUE TITLE
yaml_info decorator breaks mypy checks

### ISSUE THREAD

**Issue by MatthiasLohr** [smarie/python-yamlable#11]
Hi,

thanks for providing this library! This took a lot of headaches away from me. However, despite in the actual code everything works perfectly fine, there is a small issue woth my mypy validation.

Basically, this is my class:
```python
@yaml_info(yaml_tag_ns='myns)
class MyClass(YamlAble):
    def __init__(self, **kwargs: Any) -> None:
      # ...
      self.__dict__.update(kwargs)
```

When I now run the mypy checks, I'm getting the following error which points to the line with `class MyClass(YamlAble):`:
```
mypackage/myfile.py:29: error: Too few arguments
```

When I remove the decorator like this, the error is gone:

```python
# @yaml_info(yaml_tag_ns='myns)
class MyClass(YamlAble):
    def __init__(self, **kwargs: Any) -> None:
      # ...
      self.__dict__.update(kwargs)
```

Any idea how to solve the mypy error without losing the functionality of your library?

**Comment by Jorricks** [comment]
Did you notice you didn't close the tag:
```python
@yaml_info(yaml_tag_ns='myns)
```
compared to
```python
@yaml_info(yaml_tag_ns='myns')
```

**Comment by smarie** [comment]
Thanks @MatthiasLohr for this feedback, and @Jorricks for the solution hint ! 

I am able to reproduce your mypy error @MatthiasLohr , it happens with yamlable 1.0.3 (not earlier) as it is now declared as providing type hints.
I'll investigate.

**Comment by Jorricks** [comment]
Cool. I just started using this tool and also use MyPy so would love to figure out what the issue is.

**Comment by smarie** [comment]
The type hint for the decorator was not correct. I just fixed it, hoping that travis is still operating (I did not migrate this project yet to github actions, as I did for most others in my account)

**Comment by Jorricks** [comment]
That was quick. I was looking at the change. It's just the `StringIO as _StringIO` that mainly fixed it, correct?

**Comment by smarie** [comment]
nope :) look at the type hints for `yaml_info` rather: it was not declared as being a decorator for the appropriate kind of callable/class

**Comment by smarie** [comment]
Travis is dead, so I'll migrate to github actions now

**Comment by Jorricks** [comment]
Ahhh makes sense. Thanks for that.

**Comment by smarie** [comment]
I just released `1.0.4`, let me know if it works for you @MatthiasLohr  and @Jorricks !
